### ISSUE TITLE
Games with proton do not start

### ISSUE THREAD

**Issue by JustXLS** [ValveSoftware/Proton#7209]
#### Your system information

* Steam client version (build number or date): 1698260427
* Distribution (e.g. Ubuntu): Gentoo
* Opted into Steam client beta?: no 
* Have you checked for system updates?: yes
* Steam Logs: [steam-logs.tar.gz](https://github.com/ValveSoftware/steam-for-linux/files/13196471/steam-logs.tar.gz)

* GPU: AMD

dlmopen steamservice.so failed: steamservice.so: cannot open shared object file: No such file or directory

**Comment by kisak-valve** [comment]
Hello @JustXLS, as a quick check, can you see if `~/.local/share/Steam/ubuntu12_32/steamservice.so` exists and is about 5MB?

**Comment by JustXLS** [comment]
> Hello @JustXLS, as a quick check, can you see if `~/.local/share/Steam/ubuntu12_32/steamservice.so` exists and is about 5MB?

.rwxr-xr-x 5.5M justxls  /home/justxls/.local/share/Steam/ubuntu12_32/steamservice.so exists

**Comment by kisak-valve** [comment]
Thanks, that makes your issue either with Proton or the Steam Linux Runtime - Sniper container environment which Proton Experimental is run inside, so I've transferred this issue report to the Proton issue tracker. Please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.)

Also, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by JustXLS** [comment]
> Thanks, that makes your issue either with Proton or the Steam Linux Runtime - Sniper container environment which Proton Experimental is run inside, so I've transferred this issue report to the Proton issue tracker. Please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.)
> 
> Also, please copy the contents of Steam Runtime Diagnostics from Steam (`Steam` -> `Help` -> `Steam Runtime Diagnostics`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

i found it in the log `6197.845:0020:0118:err:winediag:nodrv_CreateWindow Application tried to create a window, but no driver could be loaded.
6197.845:0020:0118:err:winediag:nodrv_CreateWindow L"Make sure that your X server is running and that $DISPLAY is set correctly."` maybe that's the problem

**Comment by JustXLS** [comment]
https://gist.github.com/JustXLS/d3d4766dc809b50c0e0e7c5579736459 
[steam-552100.log](https://github.com/ValveSoftware/Proton/files/13196566/steam-552100.log)

**Comment by kisak-valve** [comment]
Looking through your system information, the 64 bit Vulkan render path doesn't look healthy (https://gist.github.com/JustXLS/d3d4766dc809b50c0e0e7c5579736459#file-gistfile1-txt-L180-L201). Blind guess that it's related to `Gamescope WSI (XWayland Bypass) Layer` since that's in the 64 bit Vulkan render path, but not the healthier 32 bit Vulkan render path.

Please test how the game behaves with the launch options set to `DISABLE_GAMESCOPE_WSI=1 %command%`.

**Comment by JustXLS** [comment]
> Please test how the game behaves with the launch options set to

same 
https://gist.github.com/JustXLS/b8a1593116aa23b533ca9f06a1185b24

**Comment by JustXLS** [comment]
wine in a system with dxvk works fine with xwayland

**Comment by ip-6** [comment]
Hi, I've got a similar issue - details follow.

Description: After installing a Windows game and Proton (tested with 8.0-4 and experimental) using a Linux Steam client, the game doesn't launch. No new window is shown - nothing happens, except the PLAY button changes to CANCEL, and the play time timer is running.
Console log from the Steam client doesn't show any errors.

Tested games:
[1] Tempest Rising DEMO (Proton log is for this game)
[2] Contraband Police DEMO

Logs:
[Proton log](https://gist.github.com/ip-6/e4025a22e537c686b7a5f06a223370c8)
[Runtime diagnostics](https://gist.github.com/ip-6/e1dba4082870ba342608177720b71d4c)

Laptop uses PRIME GPU offloading (dual GPU - Intel+Nvidia), but works correctly with other software (UNIGINE SuperPosition benchmark for an example). 

**Proton log** doesn't seem to show any critical errors.
"MESA-INTEL" warning messages for the performance stats are a red herring, if I understood correctly.

**Runtime diagnostics** shows a few issues:
x11/vaapi - seems that the driver cannot be found (?)
egl_x11/gl - wrong ELF class
egl_x11/glesv2 - wrong ELF class

I've originally tested with Proton Experimental and both games were not launching. Then I tried to add `DISABLE_GAMESCOPE_WSI=1 %command%` launch option and [2] was launching, while [1] didn't. After switching to Proton 8.0-4, [2] worked without the launch option, while [1] still did not.

EDIT: See my [update below](https://github.com/ValveSoftware/Proton/issues/7209#issuecomment-1868506411).

**Comment by ip-6** [comment]
@kisak-valve Please check my case, is it the same as reported by JustXLS or should it be in a separate ticket?
Thanks.

**Comment by pokemonpasta** [comment]
I seem to be running into a similar issue too. On Kubuntu 20.04, trying to play Dragon Quest XI S: Echoes of an Elusive Age Definitive Edition, and it gets stuck while launching and eventually gives up. What makes it stranger is other games being run through proton are perfectly fine. Main one I've tried is Hades, but I have some other games I could load up that I doubt will give me any issues.

I've tried everything in this thread, as well as `WINE_FULLSCREEN_FSR=1 gamemoderun %command%`, various proton versions, and updating both graphics drivers and wine, none of which seem to have helped. I'm happy to try anything else or provide more information to help resolve this issue.

[System Info](https://pastebin.com/hp4shn2s)
[Proton Log](https://pastebin.com/Avd2eKJi)
[Steam terminal output when trying to launch game](https://pastebin.com/3jnCBZ9b)

**Comment by ip-6** [comment]
After spending a day in troubleshooting and investigating (on GPU technologies available nowadays in Linux) it finally works.

**TL;DR; Upgrade Vulkan to 1.3!**

Error I was mostly investigating (in the Proton log) was:
`00bc:err:xrandr:xrandr14_get_adapters Failed to get adapters`

To troubleshoot I installed [vulkan-tools](https://github.com/KhronosGroup/Vulkan-Tools) and noticed that vkcube doesn't work out of the box. Applying one of the following made it work:
* Renaming all ICD profiles (/usr/share/vulkan/icd.d/) to *.backup except the needed one (nvidia_icd.json)
* Using an environment variable (VK_ICD_FILENAMES=/usr/share/vulkan/icd.d/nvidia_icd.json) in order to force an ICD profile

Doing the same for the Proton game didn't help, it still wasn't launching - but all the errors were gone from the [Proton log](https://gist.github.com/ip-6/e4025a22e537c686b7a5f06a223370c8).

Few other environment switches I've experimented with (just to see whether it will trigger different game/Proton behavior), without profit:
- DISABLE_GAMESCOPE_WSI=1
- PROTON_USE_WINED3D=1
- LIBGL_ALWAYS_SOFTWARE=1
- PROTON_NO_ESYNC=1 PROTON_NO_FSYNC=1
- __NV_PRIME_RENDER_OFFLOAD=1

Next stop was to investigate [NVIDIA driver](https://developer.nvidia.com/vulkan-driver) (535) and `vulkaninfo --summary` output. There I've noticed an important part - that the latest Vulkan (1.3) features are available in the latest driver (535).
Ubuntu 20.04 LTS comes with and older Vulkan (1.2). Then I've also noticed the Ubuntu 22.04 LTS release notes for MESA:
> Inclusion of mesa 22. This is really great news for Gamers – the latest vulkan stuff comes automatically..

It was obvious, that in order to activate latest NVIDIA driver features, new Vulkan version is needed - which in my case (Ubuntu) meant dist-upgrade.
After upgrading to Ubuntu 22.04 LTS, Vulkan 1.3 was automatically installed and everything worked out of the box - without any extra parameters.

p.s. Hopefully people Googling will find this helpful, and spend their time playing rather than troubleshooting :) Marry Xmas!

**Comment by pokemonpasta** [comment]
> **TL;DR; Upgrade Vulkan to 1.3!**

Updated to Kubuntu 22.04 (and thus updated vulkan) and can confirm, game is working perfectly now. Thank you!
